### PR TITLE
Skip writing to unedited files

### PR DESCRIPTION
**Pull request by phanimarupaka**
@pwittrock 

This PR is to skip writing to files which are not modified during set operation.

### CODE REVIEW

File: kyaml/setters2/settersutil/fieldsetter.go [hunk 1]
```diff
@@ -57,12 +60,49 @@ func (fs FieldSetter) Set(openAPIPath, resourcesPath string) (int, error) {
...
-		Filters: []kio.Filter{kio.FilterAll(s)},
+		Filters: []kio.Filter{FilterAll(s)},
 		Outputs: []kio.Writer{inout},
 	}.Execute()
 	return s.Count, err
 }
+
+// FilterAll applies the set filter for all yaml nodes and only returns the nodes whose
+// corresponding file has at least one node with input setter
+func FilterAll(s *setters2.Set) kio.Filter {
```
Comment (pwittrock) [inline kyaml/setters2/settersutil/fieldsetter.go]: make this private, no reason to export it

File: kyaml/setters2/settersutil/fieldsetter.go [hunk 2]
```diff
@@ -57,12 +60,49 @@ func (fs FieldSetter) Set(openAPIPath, resourcesPath string) (int, error) {
...
+	return kio.FilterFunc(func(nodes []*yaml.RNode) ([]*yaml.RNode, error) {
+		var setFiles sets.String = map[string]interface{}{}
```
Comment (pwittrock) [inline kyaml/setters2/settersutil/fieldsetter.go]: setFiles -> filesToUpdate

File: kyaml/setters2/settersutil/fieldsetter.go [hunk 3]
```diff
@@ -57,12 +60,49 @@ func (fs FieldSetter) Set(openAPIPath, resourcesPath string) (int, error) {
...
+		var setNodes []*yaml.RNode
```
Comment (pwittrock) [inline kyaml/setters2/settersutil/fieldsetter.go]: move this to line 97, it should be declared close to where it is used

File: kyaml/setters2/settersutil/fieldsetter.go [hunk 4]
```diff
@@ -57,12 +60,49 @@ func (fs FieldSetter) Set(openAPIPath, resourcesPath string) (int, error) {
 	}
 
 	// Update the resources with the new value
-	inout := &kio.LocalPackageReadWriter{PackagePath: resourcesPath}
+	inout := &kio.LocalPackageReadWriter{PackagePath: resourcesPath, NoDeleteFiles: true}
```
Comment (pwittrock) [inline kyaml/setters2/settersutil/fieldsetter.go]: add a comment about why NoDeleteFiles

File: kyaml/setters2/settersutil/fieldsetter.go [hunk 5]
```diff
@@ -57,12 +60,49 @@ func (fs FieldSetter) Set(openAPIPath, resourcesPath string) (int, error) {
...
+		var setNodes []*yaml.RNode
+		// for each node record the set fields count before and after filter is applied and
+		// store the corresponding file paths if there is an increment in setters count
+		for i := range nodes {
+			preCount := s.Count
+			_, err := s.Filter(nodes[i])
+			if err != nil {
+				return nil, errors.Wrap(err)
+			}
+			postCount := s.Count
```
Comment (pwittrock) [inline kyaml/setters2/settersutil/fieldsetter.go]: you don't need to save postCount, you can just do preCount < s.Count

File: kyaml/setters2/settersutil/fieldsetter.go (hunk 4 above)
Comment (phanimarupaka) [reply kyaml/setters2/settersutil/fieldsetter.go]: Done

File: kyaml/setters2/settersutil/fieldsetter.go (hunk 1 above)
Comment (phanimarupaka) [reply kyaml/setters2/settersutil/fieldsetter.go]: Done

File: kyaml/setters2/settersutil/fieldsetter.go (hunk 2 above)
Comment (phanimarupaka) [reply kyaml/setters2/settersutil/fieldsetter.go]: Done

File: kyaml/setters2/settersutil/fieldsetter.go (hunk 3 above)
Comment (phanimarupaka) [reply kyaml/setters2/settersutil/fieldsetter.go]: Done

File: kyaml/setters2/settersutil/fieldsetter.go (hunk 5 above)
Comment (phanimarupaka) [reply kyaml/setters2/settersutil/fieldsetter.go]: Yes.

**Pull request merged**